consensus/view_change.go

```diff
@@ -140,10 +147,36 @@ func (consensus *Consensus) getNextViewID() (uint64, time.Duration) {
 	return nextViewID, viewChangeDuration
 }
 
+type nextLeaderParams struct {
+	config    *params.ChainConfig
+	curHeader *block.Header
+}
+
+func (a nextLeaderParams) Config() *params.ChainConfig {
+	return a.config
+}
+
 // getNextLeaderKey uniquely determine who is the leader for given viewID
 // It reads the current leader's pubkey based on the blockchain data and returns
 // the next leader based on the gap of the viewID of the view change and the last
 // know view id of the block.
```

Comment:
can you review the comments here and update accordingly ? Or move that to getNextLeaderKey and have a new one for getNextLeaderKeySkipSameAddress

Reply:
updated